io/plugins/eu.esdihumboldt.hale.io.topojson/src/eu/esdihumboldt/hale/io/topojson/TopoJsonInstanceWriter.java

```diff
@@ -169,10 +181,110 @@ private List<URI> writeIntermediateShapefiles(InstanceCollection instances,
 
 	private void convertShapefileToTopoJson(String shapefilePath,
 			CoordinateReferenceSystem sourceCrs, String targetPath, String topologyName, int kink,
-			int quantizeDigit, boolean compressOutput) throws IOException {
+			int quantizeDigit, boolean compressOutput) throws Exception {
 
-		TopojsonApi.shpToTopojsonFile(shapefilePath, sourceCrs, targetPath, topologyName, kink,
+		String aJson = TopojsonApi.shpToTopojson(shapefilePath, sourceCrs, topologyName, kink,
 				quantizeDigit, compressOutput);
+		try {
+			// Parse JSON string
+			ObjectMapper mapper = new ObjectMapper();
+			JsonNode jsonNode = mapper.readTree(aJson);
+
+			// Replace "\u0000" values with null recursively
+			replaceUnwantedCharacters(jsonNode);
+
+			// Convert modified JSON node back to string
+			String modifiedJsonString = mapper.writeValueAsString(jsonNode);
+
+			if (compressOutput) {
+				Toolbox.writeFile(targetPath, Compress.compressB64(modifiedJsonString));
+			}
+			else {
+				Toolbox.writeFile(targetPath, modifiedJsonString);
+			}
+
+		} catch (Exception e) {
+			throw new Exception(e);
+		}
+	}
+
+	/**
+	 * @param node to replace the unwanted characters
+	 * @return the clean node
+	 */
+	public static JsonNode replaceUnwantedCharacters(JsonNode node) {
+		if (node.isObject()) {
+			ObjectNode objectNode = (ObjectNode) node;
+			objectNode.fields().forEachRemaining(entry -> {
+				JsonNode value = entry.getValue();
+				if (value.isTextual()) {
+
+					String textValue = value.textValue();
+					if (textValue.contains("\u0000")) {
+						if (textValue.replaceAll("\u0000", "").isEmpty()) {
+							objectNode.put(entry.getKey(), (String) null);
+						}
+						else {
+							objectNode.put(entry.getKey(), textValue.replaceAll("\u0000", ""));
+						}
+					}
+					else {
+						// Step 1: Parse the string into java.util.Date
+						try {
+							Date date = parseStringToDate(textValue);
+
+							if (date != null) {
+								// Step 2: Convert java.util.Date to
+								// java.time.Instant
+								Instant instant = date.toInstant();
+
+								// Step 3: Convert java.time.Instant to
+								// java.time.LocalDateTime
+								LocalDateTime localDateTime = instant.atZone(ZoneId.systemDefault())
+										.toLocalDateTime();
+
+								// Get the year from LocalDateTime
+								int year = localDateTime.getYear();
+								if (year == 2) {
+									objectNode.put(entry.getKey(), (String) null);
+								}
+								else {
+									objectNode.put(entry.getKey(), textValue);
+								}
+
+							}
+							else {
+								objectNode.put(entry.getKey(), textValue);
+							}
+
+						} catch (Exception e) {
+							objectNode.put(entry.getKey(), (String) null);
```

Comment:
I think it would be good to log this exception, since it seems to be unexpected that it may happen (since date parsing errors are handled somewhere else).